btcjson/chainsvrcmds.go

```diff
@@ -163,6 +163,24 @@ func NewGetBlockCountCmd() *GetBlockCountCmd {
 	return &GetBlockCountCmd{}
 }
 
+// GetBlockFilterCmd defines the getblockfilter JSON-RPC command.
+type GetBlockFilterCmd struct {
+	BlockHash  string  // The hash of the block
+	FilterType *string // The type name of the filter, default=basic
```

Comment:
Looking at the Bitcoin Core codebase, "basic" is the only known filter type there. Are there any other known types out in the wild? Might make sense to create a custom type for this instead of passing in a string.

Reply:
Haven't seen others than types `basic`. Not sure how standard the filter names are but first see 
my [previous response](https://github.com/btcsuite/btcd/pull/1579#discussion_r429965540) about doing validation only server-side.

Reply:
I can see that for instance in `btcjson.AddNodeCmd` there is a custom type `AddNodeSubCmd` being used but in that case the options are [in the spec](https://bitcoincore.org/en/doc/0.19.0/rpc/network/addnode/).

In case of getblockfilter [the spec](https://bitcoincore.org/en/doc/0.19.0/rpc/blockchain/getblockfilter/) only mentions `basic` which is the default anyway.

Reply:
[BIP 157](https://github.com/bitcoin/bips/blob/master/bip-0157.mediawiki#specification) describes that initial filter types are defined in [BIP 158](https://github.com/bitcoin/bips/blob/master/bip-0158.mediawiki#block-filters) which defines only the `Basic` filter.

Bitcoin Core CLI seems to allow describing any `getblockfilter` in a CLI command and will send the request server which will reject any unknown filter type name. I'd have the similar behavior in `btcctl` and just handle this as a string.

In `rpcclient` the second parameter of `rpcclient.GetBlockFilter` could have a more specific type, for instance we can call it `FilterTypeName`. For the server-side implementation of `getblockfilter` to resolve the filter for the name we can map the basic `FilterTypeName` to `wire.GCSFilterRegular`.

I'll look at how to best implement `FilterTypeName`. It seems a good idea for `rpcclient`.

Reply:
Good points about validation, as well as that in this specific scenario it might not be possible to construct a good type.

Just so you understand my reasoning: custom types are (at least for me) very beneficial when coding with `btcd` packages as a library, because they limit the scope of possibilities and place you on a narrower path of "valid actions". Does this make sense?

Reply:
Agreed definitely should have types for the `rpcserver` API. Sorry about the fuss - with the way you suggested the CLI mechanism works just as well (facepalm)

Please see the new commit 9d7bd87